narwhals/utils.py

```diff
@@ -1275,16 +1276,24 @@ def dtype_matches_time_unit_and_time_zone(
     )
 
 
+def _hasattr_static(obj: Any, attr: str) -> bool:
+    try:
+        getattr_static(obj, attr)
+    except AttributeError:
+        return False
+    return True
```

Comment:
Not sure if you like it better but if we use a default value then:

```suggestion
    return getattr_static(obj, attr, None) is not None
```

or 

```suggestion
    return bool(getattr_static(obj, attr, None))
```

Reply:
What if the attribute exists and returns None? Not that I would expect our uses of `'__narwhals_*__'` to ever return `None`, but it would end up creating an edgecase. If you want to avoid the try/except we can create a short-lived sentinel of our own and check for that.

```python
from inspect import getattr_static

def hasattr_static(obj, attr):
    sentinel = object()
    return getattr_static(obj, attr, sentinel) is not sentinel

class T:
    x = None

print(
    bool(getattr_static(T, "x", None)), # False; though x exists as an attribute!
    hasattr_static(T, "x"),             # True; which is correct!
    sep='\n'
)
```

Reply:
Sorry I'm late to this, but I'd vote for the `sentinel = object()` version

This is a pretty hot code path and for `<3.11` you'll be adding overhead for the exception handling https://bugs.python.org/issue40222

Reply:
I'll open up a PR with this change and we can have further discussion there!